WordPress/Classes/ViewRelated/Post/PostCardStatusViewModel.swift

```diff
@@ -56,15 +56,30 @@ class PostCardStatusViewModel: NSObject {
         if MediaCoordinator.shared.isUploadingMedia(for: post) {
             return NSLocalizedString("Uploading media...", comment: "Message displayed on a post's card while the post is uploading media")
         } else if post.isFailed {
-            if canCancelAutoUpload {
-                return StatusMessages.postWillBePublished
+            let defaultFailedMessage = StatusMessages.uploadFailed
+            guard let postStatus = post.status else {
+                return defaultFailedMessage
             }
 
             if post.wasAutoUploadCancelled {
                 return StatusMessages.localChanges
             }
 
-            return StatusMessages.uploadFailed
+            if autoUploadInteractor.autoUploadAction(for: post) != .upload {
+                return defaultFailedMessage
+            }
+            if post.hasRemote() {
+                return StatusMessages.changesWillBeUploaded
+            }
+
+            switch postStatus {
```

Comment:
This method is getting bigger, maybe we can extract the failure status to its own property? Maybe `failedStatus` or something like that.

Reply:
I create a method instead (adb140b380549637a0aa7cf957ef186ad5f02428). Adding another property makes it harder to determine how the class should be used. What do you think?